src/Autofac/Autofac.csproj

```diff
@@ -4,7 +4,7 @@
     <Description>Autofac is an IoC container for Microsoft .NET. It manages the dependencies between classes so that applications stay easy to change as they grow in size and complexity.</Description>
     <VersionPrefix>4.8.1</VersionPrefix>
     <TargetFrameworks>netstandard1.1;net45</TargetFrameworks>
-    <NoWarn>$(NoWarn);CS1591</NoWarn>
+    <NoWarn>$(NoWarn);CS1591;CA1819</NoWarn>
```

Comment:
[CA1819 indicates that public/protected properties shouldn't return arrays.](https://msdn.microsoft.com/en-us/library/0fss9skc.aspx) The reason they note is that if the property doesn't return a _copy_ of the array then people can (and likely will) start using that property as an indexed property to start poking data into places they possibly shouldn't.

I'm not a big stickler for stuff like this, but I feel like there's a decent point here - where we may have public properties (like in `IComponentRegistration`) that were returning `IEnumerable<T>` and the switch to `T[]` enables people to do things they maybe shouldn't be doing. I've read too many Raymond Chen posts where people write programs using APIs that were never intended for a particular purpose, but now it's all locked in.

I'd recommend not ignoring this and, instead, being very thoughtful about switching away from an interface and using `[SuppressMessage]` to pinpoint the specific choices and document why that choice was made.

From a larger perspective, moving from `IList<T>` to `List<T>`, from `IEnumerable<T>` to `T[]`... that stuff may limit our ability to change underlying implementations. For example, if we need to store things in a `SortedList<T>` or start using `Span<T>` in certain places, switching from an interface to a concrete type makes that implementation detail a breaking change.